v2/devices/FP4.yml

```diff
@@ -0,0 +1,198 @@
+    image: "phone_power_down"
+    button: true
+  recovery:
+    title: "Reboot to Recovery"
+    description: 'With the device still at the "FastBoot Mode", use the Volume buttons to switch to "Recovery Mode" and push the power button to confirm your selection.'
+    image: "phone_power_up"
+    button: true
+  fastbootd:
+    title: "Reboot to fastbootd mode"
+    description: 'With the device still at the "FastBoot Mode", use the Volume buttons to switch to "Recovery Mode" and push the power button to confirm your selection. Then select "Fastboot mode" from the list.'
+    image: "phone_power_up"
+    button: true
+unlock:
+  - "confirm_model"
+  - "confirm_os"
+handlers:
+  bootloader_locked:
+    actions:
+      - fastboot:oem_unlock:
+          code_url: "https://www.fairphone.com/en/bootloader-unlocking-code-for-fairphone-3"
+operating_systems:
+  - name: "Ubuntu Touch"
+    compatible_installer: ">=0.9.2-beta"
+    options:
+      - var: "channel"
+        name: "Channel"
+        tooltip: "The release channel"
+        link: "https://docs.ubports.com/en/latest/about/process/release-schedule.html"
+        type: "select"
+        remote_values:
+          systemimage:channels:
+      - var: "wipe"
+        name: "Wipe Userdata"
+        tooltip: "Wipe personal data. This is necessary for first time installations."
+        type: "checkbox"
+      - var: "partition"
+        name: "Partitioning"
+        tooltip: "Changes partition sizes as required. Only necessary for first time installations."
+        type: "checkbox"
+        value: true
+      - var: "bootstrap"
+        name: "Bootstrap"
+        tooltip: "Flash system partitions using fastboot"
+        type: "checkbox"
+        value: true
+    prerequisites: []
+    steps:
+      - actions:
+          - core:download:
+              group: "firmware"
+              files:
+                - url: "https://files.beidl.guru/fp4/boot.img"
+                  checksum:
+                    sum: "c1ab7a12e48edddc089cd93265e475ff58242919427520a589fc19203c8fb628"
+                    algorithm: "sha256"
+                - url: "https://files.beidl.guru/fp4/dtbo.img"
+                  checksum:
+                    sum: "68ad83e20eb2b47592c2b7fe48161893770b1727c7b7ce7240504eaf8612d733"
+                    algorithm: "sha256"
+                - url: "https://files.beidl.guru/fp4/recovery.img"
+                  checksum:
+                    sum: "a09ed000f16db6a61e5fa0971b9e43bdd6a2b9d07960cbb22909720ba718899b"
+                    algorithm: "sha256"
+        condition:
+          var: "bootstrap"
+          value: true
+      - actions:
+        fallback:
+          - core:user_action:
+              action: "bootloader"
+        condition:
+          var: "partition"
+          value: true
+      - actions:
+        fallback:
+          - core:user_action:
+              action: "recovery"
+        condition:
+          var: "partition"
+          value: true
+      - actions:
```

Comment:
...

but seeing as you try to reboot to recovery and into fastbootd from there, you could reboot to fastbootd directly instead:

```
      - actions:
          - fastboot:reboot_fastboot:
```